Tools/CISupport/Shared/de-duper

```diff
@@ -27,6 +27,7 @@ import os
 import sys
 
 CHUNK_SIZE = 8192
+MAX_LINKS = 64990
```

Comment:
Given this varies between filesystems, I'd rather we didn't hardcode this (and, e.g., APFS (or maybe Darwin generally) seems to have a lower limit, and NTFS has an even lower limit).

`os.pathconf(path, "PC_LINK_MAX")` should give the number of links `path` can have to it, on all POSIX-like systems (and we probably don't care about non-POSIX-like systems here).